### DISCUSSION TITLE
[Gatsby] Gatsby GraphQL and navigation for universal app

### DISCUSSION THREAD

**Discussion by lucido-apps** [expo/expo#6741] [Archived: Questions]
Hi Expo community!

I'm trying out the gatsby + expo example and evaluating this combination as a possibility for an app. I love Gatsby and Expo, so it would be wonderful to use them together to create an universal app.

However, I'm not sure how easy or hard it would be to share code between Gatsby and native versions of the app. A main feature of Gatsby is GraphQL queries used at build time to get images, data, etc. to be statically rendered. **What would happen with a component that uses Gatsby GraphQL queries? it won't be able to be used by the native app, correct?**

Also, Gatsby has it's own way of doing page based navigation. I'm wondering how feasible it would be to manage, for example, react-navigation for native and Gatsby navigation for web. **Would this add excessive complication to the code base?**

Finally, considering these two points. **Do you think it makes sense to build an universal app with Expo + Gatsby? or would it lead to over-engineering, endless configurations, two versions of the majority of components for Gatsby and native?**

Your opinions would be extremely valuable to me at this point! thank you very much!

Santi

**Comment by nandorojo** [comment]
I'm currently working on [`expo-gatsby-navigation`](https://github.com/nandorojo/expo-gatsby-navigation) to handle the navigation component. 

It integrates `react-navigation` with gatsby's `src/page` architecture and lets you use the same navigation API you normally would use with `react-navigation`. 

`expo-gatsby-navigation` will be similar to [`expo-next-react-navigation`](https://github.com/nandorojo/expo-next-react-navigation), which is already usable thanks to @EvanBacon.

I don't have much experience using gatsby + expo yet, but I can say that the Next.js + Expo integration is really awesome.
